### ISSUE TITLE
Seastar required protobuf 3.3.0, but Ubuntu 18.04 only provides 3.0.0.

### ISSUE THREAD

**Issue by vladzcloudius** [scylladb/scylla#4207]
*Installation details*
Scylla version: e2e25720c17d434ce5d4229a11ddcfb5fc17060b
OS: Ubuntu 18.04.1 LTS (packages up to date on Feb 8 2019)

*Descritption* 
I'm no longer able to compile scylla.

*Steps to reproduce*
1. Clone scylla and its modules:
```
git clone https://github.com/scylladb/scylla.git
cd scylla
git submodule update --init --recursive
```
2. Install dependencies:
```
sudo ./install-dependencies.sh
```
3. Run a configure.py:
```
./configure.py --cflags=-Wno-attributes  --mode=release
```
It fails. Here is the output.
[scylla-configure.log](https://github.com/scylladb/scylla/files/2845898/scylla-configure.log)

I tried to run a `seastar/cooking.sh` before `configure.py` but it didn't help either.

**Comment by vladzcloudius** [comment]
@slivne @hakuch FYI

**Comment by vladzcloudius** [comment]
@slivne Did we do any changes in the serialization code recently that requires the upgrade for the `protobuf` library from 3.0.0 to 3.3.0?

**Comment by avikivity** [comment]
@vladzcloudius please attach less an inline more

I'm sure it's the new buid system being overly conservative and that we can build with much older versions of protobuf. Please check if it works on your version and update the requirement.

**Comment by vladzcloudius** [comment]
@avikivity Sure. It's just that I didn't know what could be useful so I attached the whole output and it prints a lot indeed.
The interesting part of it (as I think) is as follows:

```-- Found hwloc: /usr/lib/x86_64-linux-gnu/libhwloc.so (found suitable version "1.11.9", minimum required is "1.11.2") 
-- Found lksctp-tools: /usr/lib/x86_64-linux-gnu/libsctp.so  
-- Found numactl: /usr/lib/x86_64-linux-gnu/libnuma.so  
-- Found ragel: /usr/bin/ragel (Required is at least version "6.10") 
-- Found rt: -lrt  
-- Found yaml-cpp: /usr/lib/x86_64-linux-gnu/libyaml-cpp.so (found suitable version "0.5.2", minimum required is "0.5.1") 
-- Configuring done
-- Generating done
-- Build files have been written to: /home/vladz/work/scylla/seastar/build/dev
['./cooking.sh', '-i', 'fmt', '-d', '/home/vladz/work/scylla/seastar/build/release', '--', '-DCMAKE_BUILD_TYPE=RelWithDebInfo', '-DCMAKE_C_COMPILER=gcc', '-DCMAKE_CXX_COMPILER=g++', '-DCMAKE_INSTALL_PREFIX=/usr/local', '-DSeastar_EXCLUDE_TESTS_FROM_ALL=no', '-DSeastar_EXCLUDE_APPS_FROM_ALL=no', '-DSeastar_EXCLUDE_DEMOS_FROM_ALL=no', '-DSeastar_CXX_FLAGS=-Wno-attributes;-Wno-error;-march=westmere;-O3;--param;inline-unit-growth=300;-gz;-g', '-DSeastar_LD_FLAGS=', '-DSeastar_CXX_DIALECT=gnu++17', '-DSeastar_DPDK=no', '-DSeastar_HWLOC=yes', '-DSeastar_GCC6_CONCEPTS=no', '-DSeastar_ALLOC_FAILURE_INJECTOR=no', '-DSeastar_EXCEPTION_SCALABILITY_WORKAROUND=yes', '-DSeastar_ALLOCATOR_PAGE_SIZE=no', '-DSeastar_STD_OPTIONAL_VARIANT_STRINGVIEW=1', '-DSeastar_SPLIT_DWARF=no', '-DSeastar_EXPERIMENTAL_COROUTINES_TS=no']
['./cooking.sh', '-i', 'fmt', '-d', '/home/vladz/work/scylla/seastar/build/debug', '--', '-DCMAKE_BUILD_TYPE=Debug', '-DCMAKE_C_COMPILER=gcc', '-DCMAKE_CXX_COMPILER=g++', '-DCMAKE_INSTALL_PREFIX=/usr/local', '-DSeastar_EXCLUDE_TESTS_FROM_ALL=no', '-DSeastar_EXCLUDE_APPS_FROM_ALL=no', '-DSeastar_EXCLUDE_DEMOS_FROM_ALL=no', '-DSeastar_CXX_FLAGS=-Wno-attributes;-Wno-error;-march=westmere', '-DSeastar_LD_FLAGS=', '-DSeastar_CXX_DIALECT=gnu++17', '-DSeastar_DPDK=no', '-DSeastar_HWLOC=yes', '-DSeastar_GCC6_CONCEPTS=no', '-DSeastar_ALLOC_FAILURE_INJECTOR=no', '-DSeastar_EXCEPTION_SCALABILITY_WORKAROUND=yes', '-DSeastar_ALLOCATOR_PAGE_SIZE=no', '-DSeastar_STD_OPTIONAL_VARIANT_STRINGVIEW=1', '-DSeastar_SPLIT_DWARF=no', '-DSeastar_EXPERIMENTAL_COROUTINES_TS=no']
['./cooking.sh', '-i', 'fmt', '-d', '/home/vladz/work/scylla/seastar/build/dev', '--', '-DCMAKE_BUILD_TYPE=Dev', '-DCMAKE_C_COMPILER=gcc', '-DCMAKE_CXX_COMPILER=g++', '-DCMAKE_INSTALL_PREFIX=/usr/local', '-DSeastar_EXCLUDE_TESTS_FROM_ALL=no', '-DSeastar_EXCLUDE_APPS_FROM_ALL=no', '-DSeastar_EXCLUDE_DEMOS_FROM_ALL=no', '-DSeastar_CXX_FLAGS=-Wno-attributes;-Wno-error;-march=westmere', '-DSeastar_LD_FLAGS=', '-DSeastar_CXX_DIALECT=gnu++17', '-DSeastar_DPDK=no', '-DSeastar_HWLOC=yes', '-DSeastar_GCC6_CONCEPTS=no', '-DSeastar_ALLOC_FAILURE_INJECTOR=no', '-DSeastar_EXCEPTION_SCALABILITY_WORKAROUND=yes', '-DSeastar_ALLOCATOR_PAGE_SIZE=no', '-DSeastar_STD_OPTIONAL_VARIANT_STRINGVIEW=1', '-DSeastar_SPLIT_DWARF=no', '-DSeastar_EXPERIMENTAL_COROUTINES_TS=no']
Package '/seastar' requires 'protobuf >= 3.3.0' but version of protobuf is 3.0.0
Traceback (most recent call last):
  File "./configure.py", line 1077, in <module>
    seastar_cflags, seastar_libs = query_seastar_flags(pc[mode], link_static_cxx=args.staticcxx)
  File "./configure.py", line 1068, in query_seastar_flags
    cflags = pkg_config(pc_file, '--cflags', '--static')
  File "./configure.py", line 100, in pkg_config
    output = subprocess.check_output(['pkg-config'] + list(options) + [package])
  File "/usr/lib/python3.6/subprocess.py", line 336, in check_output
    **kwargs).stdout
  File "/usr/lib/python3.6/subprocess.py", line 418, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['pkg-config', '--cflags', '--static', 'seastar/build/release/seastar.pc']' returned non-zero exit status 1.

```

I tried to hack it. The only place it seems to be mentioning it is `seastar/cmake/SeastarConfig.cmake.in` but changing it didn't seem to help. I must be missing something here.

**Comment by slivne** [comment]
@avikivity / @hakuch any idea ?

**Comment by avikivity** [comment]
Package '/seastar' requires 'protobuf >= 3.3.0' but version of protobuf is 3.0.0

**Comment by hakuch** [comment]
@vladzcloudius, the version of protobuf required by Seastar was not updated correctly in a previous patch (6da35277c6e0b05eca8493be342b842438fc6b90), and this bug should be fixed by the series on the Seastar mailing list titled "Clean-up dependency versions" which has not yet been merged.

**Comment by hakuch** [comment]
This should be fixed with a3de5581ce4b934f9a56d7d021d3dfcb3c4c0782.

**Comment by amnonh** [comment]
If protobuf is only used for Prometheus 1.8, we can drop that support

**Comment by slivne** [comment]
fixed by: 74ac1deee11cb16f81230088c7f1004f9878d7ef
